groups/bsl/bslstl/bslstl_ratio.t.cpp

```diff
@@ -0,0 +1,349 @@
+// bslstl_ratio.t.cpp                                                -*-C++-*-
+#include <bslstl_ratio.h>
+
+#include <bsls_assert.h>
+#include <bsls_asserttest.h>
+#include <bsls_bsltestutil.h>
+
+#include <stdio.h>     // 'sprintf', 'snprintf' [NOT '<cstdio>', which does not
+                       // include 'snprintf']
+#include <stdlib.h>    // 'atoi'
+
+using namespace BloombergLP;
+
+//=============================================================================
+//                             TEST PLAN
+//-----------------------------------------------------------------------------
+//                              Overview
+//                              --------
+// The component under test implements a version of the ratio operations
+// introduced into the standard library in C++11 (and 14). We test to make
+// sure that the operations exist and give sane results. 
+// The tests are not exhaustive.
+//-----------------------------------------------------------------------------
+// Groups of operations tested:
+// * SI prefixes - kilo, mega, milli, etc
+// * Ratio comparisons - less/greater/equal, etc
+// ** This includes the '_v' inline variables
+// * Ratio operations - add/subtract/multiply/divide, etc.
+// ** There are no '_t' versions of these operations.
+// ----------------------------------------------------------------------------
+// [2] USAGE EXAMPLE
```

Comment:
This is not a great test driver for learning about TEST PLAN sections since there is only one (comprehensive) test case and no methods whatsoever. However, one specific rule is: for each line in test case "Testing:" sections there is a line in TEST PLAN -- and vice versa.  So, for example, typically you'd have this in TEST PLAN (test case number will vary, of course):
```
// [ 2] USAGE EXAMPLE
```
And this in case 2:
```
        // Testing:
        //   USAGE EXAMPLE
```
which we have now (except note `[2]` versus `[ 2]`, important).  So we want
```
// [ 1] SOMETHING
```
in TEST PLAN and this in case 1:
```
        // Testing:
        //   SOMETHING
```
Perhaps SOMETHING == 'ratio' HEADER VERIFICATION

Reply:
I think I've done this now.